--- /server/drivers/glk.c	2020-07-21 13:43:52.000000000 +0300
+++ /server/drivers/glk.c	2020-12-26 07:54:41.000000000 +0300
@@ -776,23 +776,23 @@
 
   /* Remap keys according to what LCDproc expects */
   switch (c) {
-    case 'V' :			// default mapping
-    case 'I' : key = "Enter";	// pyramid mapping
+    case 'h' :			// default mapping
+    case 'H' : key = "Enter";	// pyramid mapping
 	       break;
-    case 'P' :			// default mapping
-    case 'D' : key = "Left";	// pyramid mapping
+    case 'v' :			// default mapping
+    case 'V' : key = "Left";	// pyramid mapping
 	       break;
-    case 'Q' :			// default mapping
-    case 'E' : key = "Right";	// pyramid mapping
+    case 'c' :			// default mapping
+    case 'C' : key = "Right";	// pyramid mapping
 	       break;
-    case 'L' :			// default mapping
-    case 'J' : key = "Escape";	// pyramid mapping
+    case 'r' :			// default mapping
+    case 'R' : key = "Escape";	// pyramid mapping
 	       break;
-    case 'U' :			// default mapping
-    case 'C' : key = "Up";	// pyramid mapping
+    case 'j' :			// default mapping
+    case 'J' : key = "Up";	// pyramid mapping
 	       break;
-    case 'K' :			// default mapping
-    case 'H' : key = "Down";	// pyramid mapping
+    case 'm' :			// default mapping
+    case 'M' : key = "Down";	// pyramid mapping
 	       break;
     default :  break;
 
